src/cell/free.rs

```diff
@@ -0,0 +1,41 @@
+//! Internal module defining the state of a free cell.
+
+/// Represents the state of a *free* game cell.
+///
+/// Used as the associated data in the [`Cell::Free`][free_cell] variant.
+///
+/// See [the module level documentation for more][cell_module_at_free_cells].
+///
+/// [free_cell]: crate::cell::Cell::Free
+/// [cell_module_at_free_cells]: crate::cell#free-cells
+pub struct State {
+    /// An optional [`Marking`] on the free cell.
+    ///
+    /// [marking]: crate::cell::Marking
+    pub marking: Option<Marking>,
+}
+
+/// The possible markings for *free* cells.
+///
+/// See [the module level documentation for more][cell_module_at_free_cells].
+///
+/// [free_cell]: crate::cell::Cell::Free
+/// [cell_module_at_free_cells]: crate::cell#free-cells
+pub enum Marking {
+    /// A *lamp* placed in a free cell lights up other free cells in every
+    /// direction.
+    Lamp,
+
+    /// A *cross* placed in a free cell is used to signify that there
+    /// cannot be a lamp in that cell.
+    Cross,
+}
+
+struct Illumination {
+    a: bool,
+    b: bool,
+    c: bool,
+    d: bool,
+    e: bool,
+    f: bool,
+}
```

Comment:
Maybe add a function like `Illuminated_from(side: Side) -> bool`. Also think about if this should rather be implemented on the `cell::Cell` type and how exactly. Should it just return false if the cell is a `Wall` cell? Should there still be a function here, that is private though but gets used by the public function on `cell::Cell`? Should this `FreeState` even be public?

Reply:
This not moved to #10.